webapp.js

```diff
@@ -49,12 +48,7 @@ app.use(express.static('dist'));
 app.use(bodyParser.json());
 
 app.get('/api/data', function(req, res) {
```

Comment:
Generally, if you see that you have an anonymous function that does nothing but wrap another function call, the outer function is extraneous and can / should be simplified.

Maybe try moving the block ```MongoClient.connect...``` before all the endpoint definitions?